src/sage/coding/linear_rank_metric.py

```diff
@@ -588,6 +589,73 @@ def rank_weight_of_vector(self, word):
         """
         return rank_weight(word, self.sub_field())
 
+    def rank_support_of_vector(self, word, sub_field=None, basis=None):
+        r"""
+        Return the rank support of ``word`` over ``sub_field``, i.e. the  vector space over
+        ``sub_field`` generated by its coefficients.
+
+        If ``word`` is a vector over some field `\GF{q^m}`, and ``sub_field`` is a subfield of
+        `\GF{q^m}`, the function converts it into a matrix over ``sub_field``, with
+        respect to the basis ``basis``.
+
+        INPUT:
+
+        - ``word`` -- a vector over the ``base_field`` of ``self``.
+
+        - ``sub_field`` -- (default: ``None``) a sub field of the
+          ``base_field`` of ``self``; if not specified, it is the prime
+          subfield of `\GF{p}` the ``base_field`` of ``self``.
+
+        - ``basis`` -- (default: ``None``) a basis of ``base_field`` of
+          ``self`` as a vector space over ``sub_field``. If not specified,
+          the default basis is `1,\beta,\ldots,\beta^{m-1}`, where `\beta` is
+          the generator of `\GF{q^m}` given by SageMath.
+
+        EXAMPLES::
+
+            sage: G = Matrix(GF(64), [[1,1,0], [0,0,1]])
+            sage: C = codes.LinearRankMetricCode(G, GF(4))
+            sage: a = GF(64).gen()
+            sage: c = vector([a^4 + a^3 + 1, a^4 + a^3 + 1, a^4 + a^3 + a^2 + 1])
+            sage: c in C
+            True
+            sage: C.rank_support_of_vector(c)
+            Vector space of degree 6 and dimension 2 over Finite Field of size 2
+            Basis matrix:
+            [1 0 0 1 1 0]
+            [0 0 1 0 0 0]
+
+        An example with a non canonical basis::
+
+            sage: K.<a> = GF(2^3)
+            sage: G = Matrix(K, [[1,1,0], [0,0,1]])
+            sage: C = codes.LinearRankMetricCode(G)
+            sage: c = vector([a^2, a^2, 0])
+            sage: basis = [a, a+1, a^2]
+            sage: C.rank_support_of_vector(c, basis=basis)
+            Vector space of degree 3 and dimension 1 over Finite Field of size 2
+            Basis matrix:
+            [0 0 1]
+
+        TESTS::
+
+            sage: C.rank_support_of_vector(a)
+            Traceback (most recent call last):
+            ...
+            TypeError: input must be a vector
+
+            sage: C.rank_support_of_vector(c, GF(2^4))
+            Traceback (most recent call last):
+            ...
+            TypeError: the input subfield Finite Field in z4 of size 2^4 is not a subfield of Finite Field in a of size 2^3
+        """
+        if not isinstance(word, Vector):
+            raise TypeError("input must be a vector")
```

Comment:
I would write instead (but I'm fine with the current version also):
```suggestion
        word = self.ambient_space()(word)
```